mail-server/users.nix

```diff
@@ -22,36 +22,114 @@ let
   vmail_user = {
     name = vmailUserName;
     isNormalUser = false;
-    uid = vmailUIDStart;
+    uid = vmailUID;
     home = mailDirectory;
     createHome = true;
     group = vmailGroupName;
   };
 
   # accountsToUser :: String -> UserRecord
   accountsToUser = account: {
-    name = account.name;
-    isNormalUser = false;
-    group = vmailGroupName;
-    inherit (account) hashedPassword;
+    name = lib.head (lib.splitString "@" account.name);
+    domain = lib.head (lib.tail (lib.splitString "@" account.name));
+    inherit (account) hashedPassword sieveScript;
```

Comment:
for security reasons we should set the shell to `nologin`. Otherwise the standard ssh setup would allow mail users to log in via ssh.

Reply:
I agree, if these are system users they should not have a login shell. The way this is set up though, they aren't shell login accounts (they can only be used to auth with dovecot)!